src/components/classifier/SwipeCard.js

```diff
@@ -99,64 +120,116 @@ class SwipeCard extends Component {
+                    { this.props.answers[this.state.overlayAnswerIndex].label }
+                </FontedText>lop
+            </Animated.View >
 
         const pathPrefix = Platform.OS === 'android' ? 'file://' : ''
+        const dimensionsStyle = {width: this.props.subjectDisplayWidth, height: (this.props.subjectDisplayHeight - 50)}
+        const opacityStyle = { opacity: this.props.dimensionsLoaded ? 1 : 0 }
+        const { imageIndex, localUris, pagerDimensions} = this.state
         return (
-            <TouchableOpacity 
-                style={{width: this.props.subjectDisplayWidth, height: this.props.subjectDisplayHeight}} 
-                onPress={() => this.props.onPress(this.props.subject)}
-                activeOpacity={0.9}
-            >
-                <ProgressIndicatingImage
-                    onLoadEnd={() => this.setState({imageIsVisible: true})}
-                    localUri={pathPrefix + this.state.localUri}
-                    style={[styles.image, styles.imageShadow]}
-                    resizeMethod="resize" 
-                    resizeMode="contain"
-                />
-                <View style={styles.overlayContainer}>
-                    <View style={imageDimensionStyle}>
+            <View style={[styles.cardBackground, dimensionsStyle, opacityStyle]}>
+                    <View style={styles.cardContainer}>
+                        {
+                            this.props.subject.displays.length > 1 && 
+                                <VerticalPaginationBar
+                                    onScrollDownPressed={() => this.pager.scrollTo({y: (imageIndex + 1) * pagerDimensions.height})}
+                                    onScrollUpPressed={() => this.pager.scrollTo({y: (imageIndex - 1) * pagerDimensions.height})}
+                                    totalPages={this.props.subject.displays.length} 
+                                    pageIndex={imageIndex}
+                                />
+                        }
+                        <View style={styles.container} onLayout={event => this.setState({
+                                pagerDimensions: {
+                                    width: event.nativeEvent.layout.width,
+                                    height: event.nativeEvent.layout.height
+                                }
+                            })}>
+                            <VerticalViewPager
+                                bounces={false}
+                                showsVerticalScrollIndicator={false}
+                                ref={ref => this.pager = ref}
+                                key={localUris.length}
+                                // Make this happen while scrolling
+                                onScroll={({nativeEvent}) => {
+                                    this.setState({
+                                        imageIndex: Math.round(nativeEvent.contentOffset.y/pagerDimensions.height)
```

Comment:
This seems a bit odd, is the index being set here based on size proportions?

Reply:
Nope. The idea here is that the scroll view has 'pages' of content, meaning that each image in the scroll view takes up the entire viewable size of the scrollview (`pagerDimensions`). So, in order to calculate the pager index, we divide the offset by the size of pager. By rounding this number, we get the nearest page that scroll view is at.